kew.js

```diff
@@ -681,6 +681,60 @@ function allInternal(promises) {
 }
 
 /**
+ * Takes in an array of promises or literals and returns a promise that is 
+ * fulfilled by the first given promise to be fulfilled, or rejected if all of the 
+ * given promises are rejected.
+ *
+ * @param {!Array.<!Promise>} promises
+ * @return {!Promise}
+ */
+function any(promises) {
+  if (arguments.length != 1 || !Array.isArray(promises)) {
+    promises = Array.prototype.slice.call(arguments, 0)
+  }
+  return anyInternal(promises)
+}
+
+/**
+ * A version of any() that does not accept var_args
+ *
+ * @param {!Array.<!Promise>} promises
+ * @return {!Promise}
+ */
+function anyInternal(promises) {
+  if (!promises.length) return resolve(null)
+
+  var errorOutputs = []
+  var output = []
+  var finished = false
+  var promise = new Promise()
+  var counter = promises.length
+
+  for (var i = 0; i < promises.length; i += 1) {
+    if(finished) break;
+    if (!promises[i] || !isPromiseLike(promises[i])) {
+      promise.resolve(promises[i]);
+      break;
+    } else {
+      promises[i].then(replaceEl.bind(null, output, 0))
+      .then(function resolveFirstPromise() {
+            promise.resolve(output[0]);
```

Comment:
you have a race condition. resolveFirstPromise happens asynchronously,
after the loop has finished.

On Wed, Jul 15, 2015 at 1:47 PM, Ketan Bhatt <notifications@github.com>
wrote:

> In kew.js <https://github.com/Medium/kew/pull/51#discussion_r34707149>:
>
> > +
> > +  var errorOutputs = []
> > +  var output = []
> > +  var finished = false
> > +  var promise = new Promise()
> > +  var counter = promises.length
> > +
> > +  for (var i = 0; i < promises.length; i += 1) {
> > +    if(finished) break;
> > +    if (!promises[i] || !isPromiseLike(promises[i])) {
> > +      promise.resolve(promises[i]);
> > +      break;
> > +    } else {
> > +      promises[i].then(replaceEl.bind(null, output, 0))
> > +      .then(function resolveFirstPromise() {
> > +            promise.resolve(output[0]);
>
> Hi, sorry, but I didn't understand that.
> When a promise resolves, my "finished" becomes true and i break out of
> loop.
>
> What do you mean by guard and where? I took cue (read copied) from the
> allInternal function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Medium/kew/pull/51/files#r34707149>.
>